### PR TITLE
Fix torus routing

### PR DESCRIPTION
**Pull request by jjwilke**
Virtual lanes were not getting updated correctly, leading to deadlock

### CODE REVIEW

File: sstmac/hardware/router/torus_routing.cc [hunk 1]
```diff
@@ -91,7 +91,7 @@ class TorusMinimalRouter : public Router {
   void upPath(int dim, int srcX, int dstX, header* hdr) const
   {
     auto& dimensions_ = torus_->dimensions();
-    if ((srcX + 1) % dimensions_[dim] == dstX){
+    if (srcX % dimensions_[dim] == dstX){
```
Comment (jpkenny) [inline sstmac/hardware/router/torus_routing.cc:94]: Why is this mod even necessary?

File: sstmac/hardware/router/torus_routing.cc [hunk 2]
```diff
@@ -107,7 +107,7 @@ class TorusMinimalRouter : public Router {
   void downPath(int dim, int src, int dst, header* hdr) const
   {
     auto& dimensions_ = torus_->dimensions();
-    if (src == ((dst + 1) % dimensions_[dim])){
+    if (src == (dst % dimensions_[dim])){
```
Comment (jpkenny) [inline sstmac/hardware/router/torus_routing.cc:110]: same here

**Review by jpkenny — COMMENTED** [review]
I'm confused by all of these mods. Same for line 129/130. They all seem superfluous to me. Maybe I don't understand how the wrap-around routing works. Trying to do a thorough review.

File: sstmac/hardware/router/torus_routing.cc (hunk 2 above)
Comment (jjwilke) [reply sstmac/hardware/router/torus_routing.cc:110]: rightfully so, I don't know the WTH I was looking at. This is absolute nonsense.

**Review by jpkenny — CHANGES_REQUESTED** [review]
Remove superfluous mod operations at lines 94,110,129,130

**Comment by jjwilke** [comment]
mod operations removed and added needed fix for wraparound vc updates

**Review by jpkenny — APPROVED** [review]
Fixes look good. Approved.

**Pull request merged**